### DISCUSSION TITLE
why is cmd => winget install 7zip.7zip an .exe and not an .msi?

### DISCUSSION THREAD

**Discussion by hueldoeu** [microsoft/winget-cli#3393] [Q&A] [answered]
why is cmd => winget install 7zip.7zip an .exe and not an .msi?
https://sourceforge.net/p/sevenzip/discussion/45797/thread/3f550826d8/?page=1#8fff/ba69/45f0
https://sourceforge.net/p/sevenzip/discussion/45797/thread/3f550826d8/?page=1#515c

**Comment by Trenly** [comment, marked as answer]
I believe this has to do with how winget determines which installer to use, since both the MSI and EXE are available in the manifest.

Currently, there is no way to specify a preferred installer type, so winget will use the "most applicable" installer. How it determines most applicable on a fresh install, I'm not entirely sure; however, I know that it takes into account system architecture, requested install scope, locale, and OSVersion. When performing an upgrade, winget will try to match the same install technology that was previously used, so if you installed using the MSI, the upgrade should default to the MSI

**Reply by hueldoeu** [reply]
sounds good, didn't work. where can i see the content of the manifest?

**Reply by Trenly** [reply]
https://github.com/microsoft/winget-pkgs/blob/master/manifests/7/7zip/7zip/23.01/7zip.7zip.installer.yaml

**Reply by hueldoeu** [reply]
I just have tested it again! it worked!

deinstalled 7zip. downloaded the 2nd newest version of 7zip as msi https://sourceforge.net/projects/sevenzip/files/7-Zip/22.01/7z2201-x64.msi/download . installed it and upgraded it with cmd => winget upgrade 7zip.7zip with the [7z2301-x64.msi] to version 23.01 !

**Reply by denelon** [reply]
We've recently added the "--installer-type" argument as a part of the WinGet download feature, and extended it into install and upgrade.
